### PR TITLE
Chore/improve memory consumption

### PR DESCRIPTION
**Pull request by git-yogesh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,9 +42,6 @@ brew install zstd
 
 # Decompress the dataset
 zstd -d esci-s/esci.json.zst
-
-# Filter for US locale and create en_esci.json
-grep '"locale":"us"' esci.json > en_esci.json
```
Comment (Arhan13) [inline README.md:47]: @git-yogesh why did we remove this?

File: src/fetch_products.py [hunk 2]
```diff
@@ -80,60 +76,58 @@ async def validate_image_url(
     return False, {}
 
 
-async def process_category(category: str, locale: str) -> Tuple[List[Dict], List[str]]:
-    """Process products for a given category and return valid products and ASINs"""
-    valid_asins = []
+async def process_category(category: str, locale: str) -> int:
```
Comment (Arhan13) [inline src/fetch_products.py:79]: @git-yogesh valid asin's are being store in the file in the process_category function correct?

File: README.md (hunk 1 above)
Comment (git-yogesh) [reply README.md:47]: English comprises more than 60% of the original data, so directly filtering would avoid saving and managing additional file for `en`.

File: src/fetch_products.py (hunk 2 above)
Comment (git-yogesh) [reply src/fetch_products.py:79]: Yes, you're right. To avoid saving large list on memory, valid asins are being written to json file in the loop

**Review by Arhan13 — APPROVED** [review]

**Pull request merged**